### PR TITLE
IBP-4634 Move Coding to API

### PR DESCRIPTION
**Pull request by abatac**

### CODE REVIEW

File: src/main/jhipster/src/main/webapp/app/germplasm-manager/coding/germplasm-coding-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+				<div class="form-group">
+					<div class="pl-0 col-sm-12">
+						<span class="mandatory-asterisk"></span> Indicates a mandatory field
+					</div>
+				</div>
+				<div class="form-group row required">
+					<label class="col-sm-5 col-form-label font-weight-bold" jhiTranslate="germplasm-coding.coding.level"></label>
+					<div class="col-sm-7">
+						<div class="form-inline">
+							<div class="form-group">
```
Comment (darla-leafnode) [inline src/main/jhipster/src/main/webapp/app/germplasm-manager/coding/germplasm-coding-dialog.component.html:20]: Better to use Angular looping for each code name type?

File: src/main/jhipster/src/main/webapp/app/germplasm-manager/coding/germplasm-coding-result-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    templateUrl: './germplasm-coding-result-dialog.component.html'
+})
+export class GermplasmCodingResultDialogComponent implements OnInit {
+
+    results: GermplasmCodeNameBatchResultModel[];
+
+    constructor(private modal: NgbActiveModal) {
+    }
+
+    ngOnInit(): void {
```
Comment (darla-leafnode) [inline src/main/jhipster/src/main/webapp/app/germplasm-manager/coding/germplasm-coding-result-dialog.component.ts:16]: Do you need to implement `OnInit `when there's no init action to perform?

Comment (abatac) [reply src/main/jhipster/src/main/webapp/app/germplasm-manager/coding/germplasm-coding-result-dialog.component.ts]: Done.

File: src/main/jhipster/src/main/webapp/app/germplasm-manager/coding/germplasm-coding-dialog.component.html (hunk 1 above)
Comment (abatac) [reply src/main/jhipster/src/main/webapp/app/germplasm-manager/coding/germplasm-coding-dialog.component.html:20]: We only 3 predefined code names. I don't see the benefit of rendering the options in a loop.

**Review by darla-leafnode — APPROVED** [review]

**Pull request merged**